### PR TITLE
Add blend group alpha and sync clip data in clip_leaf

### PR DESCRIPTION
**Pull request by dfrg**
This adds blend group alpha via a new parameter in `SceneBuilder::push_layer`.

The more invasive change is that EndClip now carries no data in the scene buffer (removing the need to maintain a layer stack during scene building). The associated data is copied into the info buffer for BeginClip in draw_leaf. This is then sync'd up with the matching EndClip in clip_leaf. 

The coarse shader now pulls blend data from info rather than scene.

I believe I've correctly updated the draw tags to contain the appropriate bit-packed counts: for BeginClip (0x89), 2 scene words, 2 info words. For EndClip (0xa1), 0 scene words, 2 info words.

The clip_inp buffer has become an array of struct `ClipInp` that now holds the draw index and the path index, so those elements have doubled in size. This is necessary to get the `DrawMonoid` for the parent to access the info offset in clip_leaf.

### CODE REVIEW

**Review by raphlinus — APPROVED** [review]
I have a feeling this can be rearranged so that blend and alpha are retrieved from scene when needed, rather than copied through info, but haven't worked out the details. In the meantime, this does the trick, and reduces the encoding burden (which is always nice). I might take a look at that when I do rectangles, where I will reconsider some of the representation choices anyway.

Changes suggested are naming/comment, shouldn't block merging.

File: piet-wgsl/shader/coarse.wgsl [hunk 1]
```diff
@@ -371,11 +372,11 @@ fn main(
                 // In "clip zero" state, suppress all drawing
                 switch drawtag {
                     // DRAWTAG_BEGIN_CLIP
-                    case 0x05u: {
+                    case 0x89u: {
```
[the remark this answers is not in the document]
Comment (dfrg) [reply piet-wgsl/shader/coarse.wgsl:375]: This was really, really annoying.

File: piet-wgsl/shader/shared/clip.wgsl [hunk 2]
```diff
@@ -10,6 +10,11 @@ fn bic_combine(x: Bic, y: Bic) -> Bic {
     return Bic(x.a + y.a - m, x.b + y.b - m);
 }
 
+struct ClipInp {
+    ix: u32,
+    path_ix: i32,
```
[the remark this answers is not in the document]
Comment (dfrg) [reply piet-wgsl/shader/shared/clip.wgsl:15]: Will do!

File: piet-scene/src/scene/builder.rs [hunk 3]
```diff
@@ -284,8 +281,8 @@ fn encode_blend_mode(mode: BlendMode) -> u32 {
 const DRAWTAG_FILLCOLOR: u32 = 0x44;
```
[the remark this answers is not in the document]
Comment (dfrg) [reply piet-scene/src/scene/builder.rs:281]: Oh, good catch. There might be other places that require similar updates.

**Comment by raphlinus** [comment]
> Actually... what if we just set the scene_offset of EndClip to match BeginClip in clip_leaf? That should do the trick?

In the draw_monoid? Probably yes, I think so.

**Comment by dfrg** [comment]
Now I get to adjust the tags again :)

**Comment by raphlinus** [comment]
> Now I get to adjust the tags again :)

Always happy to cause trouble!

**Review by raphlinus — APPROVED** [review]
Nice! This I can get behind.

**Comment by dfrg** [comment]
Thanks for pointing out the potential for a better solution! I felt like one might be there but I missed it on the first pass :)

**Pull request merged**